nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -773,8 +773,12 @@ def build_model_parallel_config(self) -> ModelParallelConfig:
         cfg = OmegaConf.to_container(self.cfg, resolve=True)
 
         # map precision related configs
-        precision = cfg.get('precision', 32)  # PTL trainer precision
         megatron_amp_O2 = cfg.get('megatron_amp_O2', False)
+        if self.torch_dtype == torch.float16 and not self.trainer.precision.startswith("16"):
+            # Make sure that trainer.precision_plugin.scaler exists for config_mapping below
+            raise ValueError(
+                "Creating a half-precision model requires setting trainer.precision to '16' or '16-mixed'"
+            )
```

Comment:
Should we do the same check for bf16?

Reply:
I think it is needed only for `model.precision=16` which is `self.torch_dtype` here, see [this](https://github.com/NVIDIA/NeMo/blob/08937c8e7dd2e782fac99c6c230ae0170c277700/nemo/collections/nlp/models/language_modeling/megatron_base_model.py#L112). More specifically, the check is to make sure [this will work](https://github.com/NVIDIA/NeMo/blob/08937c8e7dd2e782fac99c6c230ae0170c277700/nemo/collections/nlp/models/language_modeling/megatron_base_model.py#L793-L795) -- loading FP16 model assumes that a scaler is available in `trainer` so it needs to be configured properly.

In my understanding in other cases it is the trainer precision to handle training precision if `model.precision` is different, and this would result in necessary type casting. But I might have not considered all the possibilities.

Which case exactly you mean for **model.precision** and **trainer.precision**, respectively?